packages/dashboard/src/components/sidePanel/sections/legendSettingSection/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FC, useEffect, useState } from 'react';
+import { DashboardMessages } from '../../../../messages';
+import {
+  ExpandableSection,
+  Grid,
+  Input,
+  InputProps,
+  Select,
+  SpaceBetween,
+  Toggle,
+} from '@cloudscape-design/components';
+import { useAppKitWidgetInput } from '../../utils';
+import { LEGEND_POSITION, LegendConfig } from '@synchro-charts/core';
+import ExpandableSectionHeader from '../../shared/expandableSectionHeader';
+import { SelectProps } from '@cloudscape-design/components/select/interfaces';
+
+const defaultLegendSettings: LegendConfig = {
+  legendLabels: { title: '' },
+  position: LEGEND_POSITION.BOTTOM,
+  showDataStreamColor: true,
+  width: 150,
+};
+const LegendSettings: FC<{ messageOverrides: DashboardMessages }> = ({
+  messageOverrides: {
+    sidePanel: { legendSettingMessages },
+  },
+}) => {
+  const [legendSettings, setLegendSettings] = useAppKitWidgetInput('legend');
+  const onWidthChange = ({ detail }: { detail: InputProps.ChangeDetail }) => {
+    setLegendSettings({ ...defaultLegendSettings, ...legendSettings, width: parseInt(detail.value) });
+  };
+  const [showLegend, setShowLegend] = useState<boolean>(!!legendSettings);
+  const [recentLegendSettings, setRecentLegendSettings] = useState<LegendConfig>();
```

Comment:
when user toggles back view legend, this preserves previous settings.

Reply:
I think we can improve on the pattern of blowing away every legend setting with `setLegendSettings(undefined)` below and then having to do this awkward thing where we 'remember' the previous settings on the client to prevent unexpected deletion of settings when a customer clicks the show legend toggle twice.

Why don't we just include a `shown` boolean in the legend settings object? We can completely delete `showLegend` and `recentLegendSettings` here because everything we need is always encoded in what we're dispatching to Redux.